spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -109,6 +112,58 @@ object Engine {
   private var physicalCoreNumber = -1
   private var nodeNum: Int = -1
 
+  private var gatewayServer: py4j.GatewayServer = null
+
+
+  private[bigdl] def createJavaGateway(driverPort: Int): Unit = {
+    if (SparkUtils.isDriver) {
+      val file = new java.io.File(SparkFiles.getRootDirectory(), "gateway_port")
+      if (file.exists()) {
+        file.delete()
+      }
+      file.createNewFile()
+      val out = new PrintWriter(file)
+      out.print(driverPort)
+      out.flush()
+      out.close()
+      return
+    }
+    if (gatewayServer != null) return
+    this.synchronized {
+      if (gatewayServer == null) {
+        gatewayServer = new py4j.GatewayServer(null, 0)
+      }
+    }
+    val thread = new Thread(new Runnable() {
+      override def run(): Unit = try {
+        gatewayServer.start()
```

Comment:
should we add a hooker to stop the gatewayServer once the thread is closed? i.e:
`Runtime.getRuntime().addShutdownHook`